### PR TITLE
Add TransferAssertionAction proto

### PR DESCRIPTION
**Pull request by trevormcdonald**
## Proposed change/fix

Add `TransferAssertionAction` protobuf

## Types of changes

What types of changes does this pull request introduce to ConsenSource? _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (could be a small readme update, documentation contribution, etc.)

## How to run/test

`clip`
`cargo build`
`cargo test`

### CODE REVIEW

File: protos/payload.proto [hunk 1]
```diff
@@ -218,3 +220,10 @@ message AssertAction {
       CreateStandardAction new_standard = 4;
     }
 }
+
+message TransferAssertionAction {
+    // Assertion ID of the object to have ownership changed
+    string assertion_id = 1;
+    // public key of the new owner
+    string public_key = 2;
```
Comment (Patrick-Erichsen) [inline protos/payload.proto:228]: I feel like we could settle on a common terminology for the two parties involved here. E.g. "Owner" and "Transferee". Then we could have a field that is a little more descriptive, e.g. 

```
string transferee_public_key = 2;
```

I don't love "Owner" and "Transferee" though - any better thoughts?

Comment (trevormcdonald) [reply protos/payload.proto:228]: How about `new_owner_public_key`?

Comment (Patrick-Erichsen) [reply protos/payload.proto:228]: Yeah that is simple enough haha - works for me 👍

Comment (pa3ng) [reply protos/payload.proto]: `new_owner_public_key` is easier to read imo

**Pull request merged**